katdal/chunkstore_rados.py

```diff
@@ -99,26 +99,31 @@ def from_config(cls, config, pool, keyring=None, timeout=5.):
             raise StoreUnavailable(str(e))
         return cls(ioctx)
 
-    def get(self, array_name, slices, dtype):
+    def get_chunk(self, array_name, slices, dtype):
         """See the docstring of :meth:`ChunkStore.get`."""
         dtype = np.dtype(dtype)
         key, shape = self.chunk_metadata(array_name, slices, dtype=dtype)
-        num_bytes = int(np.prod(shape)) * dtype.itemsize
+        expected_bytes = int(np.prod(shape)) * dtype.itemsize
         with self._standard_errors(key):
             # Try to read an extra byte to see if data is more than expected
-            data_str = self.ioctx.read(key, num_bytes + 1)
-        if len(data_str) != num_bytes:
+            data_str = self.ioctx.read(key, expected_bytes + 1)
+        actual_bytes = len(data_str)
+        if actual_bytes != expected_bytes:
+            # Get the actual value via stat() to improve error reporting
+            if actual_bytes > expected_bytes:
+                actual_bytes, _ = self.ioctx.stat(key)
```

Comment:
Should this be done inside `_standard_errors`, so that if it fails (e.g. because the link to the store failed, or the object got deleted in between) you get a standard exception instead of a rados one?

Reply:
Good idea, done.